chips/sam4l/src/dma.rs

```diff
@@ -144,12 +153,18 @@ impl DMAChannel {
         DMAChannel {
             registers: (DMA_BASE_ADDR + (channel as usize) * DMA_CHANNEL_SIZE) as *mut DMARegisters,
             nvic: nvic,
-            client: None,
+            client: Cell::new(None),
+            width: Cell::new(DMAWidth::Width8Bit),
             enabled: Cell::new(false),
             buffer: TakeCell::empty(),
         }
     }
 
+    pub fn initialize(&self, client: &'static mut DMAClient, width: DMAWidth) {
```

Comment:
Is there ever a case where we want to re-initialize with a new client and/or width? If so, should this perhaps return an Option with the prior DMAClient? Or should there be a way to pull the old client out? Since this is a mutable reference, if we stop it we lose it?